api/apiFunctions.test.tsx

```diff
@@ -24,294 +22,360 @@ jest.mock('./apiFunctions', () => {
   };
 });
 
-describe('Auth Functions', () => {
-  jest.mock('./apiFunctions', () => ({
-    loginAccount: jest.fn(),
-    registerAccount: jest.fn(),
-  }));
-
-  xdescribe('login account successful', () => {
-    it('should show user login successfully', async () => {
-      const userDummy = {
-        email: 'testemail@email.com',
-        password: 'test1234',
-      };
-      await loginAccount(userDummy);
-      expect(account.createEmailPasswordSession).toBeInstanceOf(Object);
+jest.mock('@/utils/getBaseUrl', () => ({
+  getBaseURL: jest.fn(),
+}));
+
+jest.mock('./config', () => ({
+  account: {
+    create: jest.fn(),
+    createRecovery: jest.fn(),
+    updateRecovery: jest.fn(),
+  },
+  ID: {
+    unique: jest.fn(),
+  },
+}));
+
+describe('apiFunctions', () => {
+  describe('Authentication Functions', () => {
+    const mockBaseURL = 'http://example.com';
+    const mockEmail = 'test@example.com';
+    const mockPassword = 'newPassword123';
+    const mockRecoveryToken = { userId: '123', secret: 'abc123' };
+    const mockToken = 'abc123';
+    const mockUserId = '123';
+    const mockUniqueId = 'unique123';
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+      (getBaseURL as jest.MockedFunction<typeof getBaseURL>).mockReturnValue(
+        mockBaseURL,
+      );
     });
 
-    //user failed to log in
-    it('should send error if user could not log in', async () => {
-      const failDummy = {
-        email: 'testemil@email.com',
-        password: 'tet1234679',
-      };
-      await loginAccount(failDummy);
-      expect(loginAccount(failDummy)).rejects.toThrow('error');
+    describe('registerAccount', () => {
+      it('should successfully register a new account', async () => {
+        const mockUser: IUser = {
+          documentId: '123',
+          id: mockUniqueId,
+          email: mockEmail,
+          leagues: [],
+        };
+
+        (ID.unique as jest.Mock).mockReturnValue(mockUniqueId);
+        (account.create as jest.Mock).mockResolvedValue(mockUser);
+
+        const result = await registerAccount({
+          email: mockEmail,
+          password: mockPassword,
+        });
+
+        expect(account.create).toHaveBeenCalledWith(
+          mockUniqueId,
+          mockEmail,
+          mockPassword,
+        );
+        expect(result).toEqual(mockUser);
+      });
+
+      it('should throw an error if registration fails', async () => {
+        const mockError = new Error('Registration failed');
+        (account.create as jest.Mock).mockRejectedValue(mockError);
```

Comment:
nit: since we're not using mockError anywhere else in this particular test, we could combine these into one line for cleaner code

Reply:
✅ fixed!